### PR TITLE
stat/distuv: Add Gumbel distribution

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
Currently the pdf is only tested with `mu=0, beta =1` because that's what scipy has, and my R installation is having issues. If anyone wants to provide other numbers I'll gladly add them.

File: stat/distuv/general.go [hunk 1]
```diff
@@ -20,5 +20,7 @@ var (
 )
 
 var (
-	expNegOneHalf = math.Exp(-0.5)
+	expNegOneHalf   = math.Exp(-0.5)
+	eulerMascheroni = 0.5772156649015328606065120900824024310421
```
Comment (kortschak) [inline stat/distuv/general.go]: These can be consts. Add `// https://oeis.org/A001620`?

File: stat/distuv/gumbel.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+// Gumbel implements the Gumbel distribution, a two-parameter continuous
+// distribution with support over the real numbers.
+//
+// The Gumbel distribution has density function
+//  1/beta * exp(-(z + exp(-z)))
+//  z = (x - mu)/beta
+// Beta must be greater than 0.
+//
+// For more information, see https://en.wikipedia.org/wiki/Gumbel_distribution .
```
Comment (kortschak) [inline stat/distuv/gumbel.go]: I think you can omit the space before the period, but please check.

File: stat/distuv/gumbel.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+// Prob computes the value of the probability density function at x.
+func (g Gumbel) Prob(x float64) float64 {
+	return math.Exp(g.LogProb(x))
+}
+
+// Quantile returns the inverse of the cumulative probability distribution.
+func (g Gumbel) Quantile(p float64) float64 {
+	if p < 0 || p > 1 {
```
Comment (kortschak) [inline stat/distuv/gumbel.go]: `if p < 0 || 1 < p {`

File: stat/distuv/general.go [hunk 4]
```diff
@@ -20,5 +20,7 @@ var (
...
+	apery           = 1.2020569031595942853997381615114499907649
```
Comment (kortschak) [inline stat/distuv/general.go]: `// https://oeis.org/A002117`

**Comment by btracey** [comment]
I think we should rename this to GumbelLeft. The wikipedia definition is the one we have here, but Wolfram's GumbelDistribution is flipped, and scipy has gumbel_l and gumbel_r .

**Comment by btracey** [comment]
or rather, GumbelRight, as this is right-skewed

File: stat/distuv/gumbel.go (hunk 3 above)
Comment (btracey) [reply stat/distuv/gumbel.go]: Done.

File: stat/distuv/general.go (hunk 1 above)
Comment (btracey) [reply stat/distuv/general.go]: Done.

File: stat/distuv/gumbel.go (hunk 2 above)
Comment (btracey) [reply stat/distuv/gumbel.go]: Done.

File: stat/distuv/general.go (hunk 4 above)
Comment (btracey) [reply stat/distuv/general.go]: Done.

**Pull request merged**